### PR TITLE
LaRomance upper bound handling

### PR DESCRIPTION
**Pull request by aruybalid**
Closes issue #20974, by removing options for cases when the input domain beyond the upper bounds is probed. Default is now EXCEPTION to allow time cut-backs and continue simulation, and USELIMIT has been removed as an option (only for the upper bounds).

Improves model trustworthiness, because surrogate model has not been calibrated outside the upper global limits and is not well-suited for extrapolation.

### CODE REVIEW

**Comment by aruybalid** [comment]
@tophmatthews, could you double check the 3tile test variations (e.g., json)?

File: modules/tensor_mechanics/test/tests/rom_stress_update/3tile.i [hunk 1]
```diff
@@ -161,10 +161,12 @@ endtime = 1.8
   [temperature]
     type = ElementAverageValue
     variable = temperature
+    # outputs = console
```
Comment (tophmatthews) [inline modules/tensor_mechanics/test/tests/rom_stress_update/3tile.i:164]: Remove comments

File: modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C [hunk 2]
```diff
@@ -68,8 +68,7 @@ LAROMANCEStressUpdateBaseTempl<is_ad>::validParams()
   MooseEnum extrapolated_lower_limit_behavior(
       "ERROR WARN IGNORE EXCEPTION DONOTHING USELIMIT EXTRAPOLATE", "EXTRAPOLATE");
   // Forbid extrapolation of on high end of limit
```
Comment (tophmatthews) [inline modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C:70]: I wonder if the option for ignore, warn, and donothing should be removed to, and only allow error or exception. I'd update the comment here too.

**Review by tophmatthews — CHANGES_REQUESTED** [review]
Looks good to me! Is there anything to be updated in the .md file about the forbidding going above the upper bound?

**Comment by tophmatthews** [comment]
I think `3tile_json_out.csv` is exactly the same as `3tile_out.csv`, right? If so, use a soft link instead of its own file.

**Comment by aruybalid** [comment]
I don't understand this (part of) error message upon run_tests:
```
test:rom_stress_update.rom/3tile_json_export:     from deepdiff import DeepDiff
test:rom_stress_update.rom/3tile_json_export: ModuleNotFoundError: No module named 'deepdiff'
```

**Comment by tophmatthews** [comment]
> I don't understand this (part of) error message upon run_tests:

Since this doesn't really show up on the civet tests, this smells like a local conda update requirement? I wonder if this is something required by what @dschwen implemented. Make sure you're using the conda python, e.g. which python

**Comment by tophmatthews** [comment]
I'll put in a fix for the bison app test failures that'll have to be fixed before this can go in.

**Comment by dschwen** [comment]
> I wonder if this is something required by what @dschwen implemented. Make sure you're using the conda python, e.g. which python

`deepdiff` is a python package that is required for our `JSONDiff` tester.

**Comment by aruybalid** [comment]
`which python` gives me:
`/Users/aruybalid/mambaforge3/envs/moose/bin/python`

And `mamba list `shows the following python version:

`python                    3.9.12          h8b4d769_1_cpython    conda-forge`

**Comment by loganharbour** [comment]
> @dschwen Can you add @aruybalid to developers so his tests automatically start?

done

File: modules/tensor_mechanics/test/tests/rom_stress_update/gold/3tile_json_export.json [hunk 3]
```diff
@@ -1 +0,0 @@
-../../../../data/laromance/test/3tile.json
```
Comment (dschwen) [inline modules/tensor_mechanics/test/tests/rom_stress_update/gold/3tile_json_export.json:1]: Please keep this a symbolic link!

Comment (tophmatthews) [reply modules/tensor_mechanics/test/tests/rom_stress_update/gold/3tile_json_export.json:1]: Yeah, this shouldn't be in here...try
```
cd modules/tensor_mechanics/test/tests/rom_stress_update/gold/
rm ./3tile_json_out.csv
ln -s ./3tile_out.csv ./3tile_json_out.csv
```

**Comment by aruybalid** [comment]
Could someone start the tests? I accepted the INL developer invite after pushing (my bad)..

File: modules/tensor_mechanics/test/tests/rom_stress_update/gold/3tile_json_export.json (hunk 3 above)
Comment (tophmatthews) [reply modules/tensor_mechanics/test/tests/rom_stress_update/gold/3tile_json_export.json:1]: So looks like this file got changed in the shuffle too, and should remain a symbolic link? I don't think you need to change it, correct?

**Review by tophmatthews — CHANGES_REQUESTED** [review]
Almost!

**Review by dschwen — CHANGES_REQUESTED** [review]
Undo symlink breakage please

File: modules/tensor_mechanics/test/tests/rom_stress_update/gold/3tile_json_export.json (hunk 3 above)
Comment (dschwen) [reply modules/tensor_mechanics/test/tests/rom_stress_update/gold/3tile_json_export.json:1]: This is still not fixed in the latest commit 13mins ago!

**Review by tophmatthews — APPROVED** [review]
🎉

**Comment by aruybalid** [comment]
Bison tests failing?

**Comment by tophmatthews** [comment]
Yeah...it'll work soon

File: modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C [hunk 4]
```diff
@@ -67,9 +67,8 @@ LAROMANCEStressUpdateBaseTempl<is_ad>::validParams()
 
   MooseEnum extrapolated_lower_limit_behavior(
       "ERROR WARN IGNORE EXCEPTION DONOTHING USELIMIT EXTRAPOLATE", "EXTRAPOLATE");
-  // Forbid extrapolation of on high end of limit
-  MooseEnum extrapolated_upper_limit_behavior("ERROR WARN IGNORE EXCEPTION DONOTHING USELIMIT",
-                                              "USELIMIT");
+  // Only allow ERROR and EXCEPTION on upper bounds
+  MooseEnum extrapolated_upper_limit_behavior("ERROR EXCEPTION", "EXCEPTION");
```
Comment (tophmatthews) [inline modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C:71]: You'll have to change make a new copy of `error_limit_behavior` for `cell_input_window_high_failure`, `wall_input_window_high_failure`, and `old_strain_input_window_high_failure` too actually...

**Review by tophmatthews — CHANGES_REQUESTED** [review]

File: modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C (hunk 4 above)
Comment (aruybalid) [reply modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C:71]: good catch!

File: modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C [hunk 5]
```diff
@@ -36,40 +36,40 @@ LAROMANCEStressUpdateBaseTempl<is_ad>::validParams()
...
+      error_upper_limit_behavior,
       "What to do if old strain is outside the upper global window of applicability.");
 
   MooseEnum extrapolated_lower_limit_behavior(
       "ERROR WARN IGNORE EXCEPTION DONOTHING USELIMIT EXTRAPOLATE", "EXTRAPOLATE");
-  // Forbid extrapolation of on high end of limit
-  MooseEnum extrapolated_upper_limit_behavior("ERROR WARN IGNORE EXCEPTION DONOTHING USELIMIT",
-                                              "USELIMIT");
+  // Only allow ERROR and EXCEPTION on upper bounds
+  MooseEnum extrapolated_upper_limit_behavior("ERROR EXCEPTION", "EXCEPTION");
```
Comment (tophmatthews) [inline modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C:72]: Just use `error_upper_limit_behavior` instead here.

**Review by tophmatthews — CHANGES_REQUESTED** [review]

**Review by tophmatthews — APPROVED** [review]
👍

**Comment by tophmatthews** [comment]
Ready for review @dschwen

**Review by dschwen — APPROVED** [review]
Looks fine now!

**Pull request merged**